amq-quarkus-school-timetabling/amq-quarkus-school-timetabling-solver/src/main/java/org/acme/schooltimetabling/messaging/TimeTableMessagingHandler.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.acme.schooltimetabling.messaging;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.acme.schooltimetabling.domain.TimeTable;
+import org.acme.schooltimetabling.message.SolverRequest;
+import org.acme.schooltimetabling.message.SolverResponse;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@ApplicationScoped
+public class TimeTableMessagingHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TimeTableMessagingHandler.class);
+    public static final String SOLVER_REQUEST_CHANNEL = "solver_request";
+    public static final String SOLVER_RESPONSE_CHANNEL = "solver_response";
+
+    Solver<TimeTable> solver;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Inject
+    @Channel(SOLVER_RESPONSE_CHANNEL)
+    Emitter<String> solverResponseEmitter;
+
+    @Inject
+    TimeTableMessagingHandler(SolverFactory<TimeTable> solverFactory) {
+        solver = solverFactory.buildSolver();
+    }
+
+    @Incoming(SOLVER_REQUEST_CHANNEL)
+    public CompletionStage<Void> solve(Message<String> solverRequestMessage) {
+        return CompletableFuture.runAsync(() -> {
+            SolverRequest solverRequest = null;
```

Comment:
(nitpick) I don't `= null` if it isn't needed (the code either fills in correctly or the catch returns or throws.
```suggestion
            SolverRequest solverRequest;
```